### PR TITLE
Fix r_core_editor() on Windows ##ports

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Test plan**

Manually tested

### CODE REVIEW

File: libr/util/str.c [hunk 1]
```diff
@@ -885,7 +885,7 @@ R_API char *r_str_append(char *ptr, const char *string) {
 	if (string && !ptr) {
 		return strdup (string);
 	}
-	if (!string) {
+	if (!string || !*string) {
```
Comment (trufae) [inline libr/util/str.c:888]: Use RSTRISEMPTY macro

File: libr/util/file.c [hunk 2]
```diff
@@ -1062,18 +1062,29 @@ R_API int r_file_mkstemp(R_NULLABLE const char *prefix, char **oname) {
...
+	LPTSTR prefix_ = r_sys_conv_utf8_to_win (pfxx);
 
 	name = (LPTSTR)malloc (sizeof (TCHAR) * (MAX_PATH + 1));
 	if (!name) {
 		goto err_r_file_mkstemp;
 	}
 	if (GetTempFileName (path_, prefix_, 0, name)) {
 		char *name_ = r_sys_conv_win_to_utf8 (name);
-		h = r_sandbox_open (name_, O_RDWR|O_EXCL|O_BINARY, 0644);
+		name_[strlen (name_) - 4] = '\0';
```
Comment (trufae) [inline libr/util/file.c:1085]: Is strlen(name_) always more than 4?

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/util/file.c (hunk 2 above)
Comment (GustavoLCR) [reply libr/util/file.c:1085]: yes, from MSDN:
```
The GetTempFileName function creates a temporary file name of the following form:

<path>\<pre><uuuu>.TMP
```

**Review by trufae — APPROVED** [review]
lets trust the windows apis

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
Looks like it broke AppVeyor:
```
[XX] C:\projects\radare2\test\db\tools\rabin2 rabin2 -qOp//8 on PE32+
ANSICON=1 R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'mkdir .tmp
rm .tmp/rabin2testpe2.exe
!rabin2 -qOp//8 -o.tmp/rabin2testpe2.exe %R2_FILE%
!rabin2 -S .tmp/rabin2testpe2.exe
rm .tmp/rabin2testpe2.exe
' bins/pe/normal64.exe
-- stdout
@@ -1,7 +1 @@
 wx 10000000 @ 0x16c
-[Sections]
-
-nth paddr        size vaddr        vsize perm name
---------------------------------------------------
-0   0x00000200  0x200 0x00401000  0x1000 s--- sect_0
-
-- stderr
Cannot open '.tmp/rabin2testpe2.exe' for writing
Cannot dump :(
r_core: Cannot open file '.tmp/rabin2testpe2.exe'
[XX] C:\projects\radare2\test\db\tools\rabin2 rabin2 -qOp/.data/rwx on PE32
ANSICON=1 R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc 'mkdir .tmp
rm .tmp/rabin2testpe1.exe
!rabin2 -qOp/.data/rwx -o.tmp/rabin2testpe1.exe %R2_FILE%
!rabin2 -S .tmp/rabin2testpe1.exe
rm .tmp/rabin2testpe1.exe
' bins/pe/test.exe
-- stdout
@@ -1,11 +1 @@
 wx e0000040 @ 0x244
-[Sections]
-
-nth paddr         size vaddr        vsize perm name
----------------------------------------------------
-0   0x00000400  0x8a00 0x00401000  0x9000 -r-x .text
-1   0x00008e00  0x2400 0x0040a000  0x3000 -r-- .rdata
-2   0x0000b200  0x1000 0x0040d000  0x3000 -rwx .data
-3   0x0000c200   0x200 0x00410000  0x1000 -r-- .rsrc
-4   0x0000c400   0xe00 0x00411000  0x1000 -r-- .reloc
-
-- stderr
Cannot open '.tmp/rabin2testpe1.exe' for writing
Cannot dump :(
r_core: Cannot open file '.tmp/rabin2testpe1.exe'
```

**Pull request merged**